keps/sig-node/4742-node-topology-downward-api/README.md

```diff
@@ -0,0 +1,416 @@
+example is utilizing topology data from cloud providers, which are automatically
+assigned to nodes, and then used to tailor workloads to minimize cross-zone data
+transfers and reduce costs. Having an easy way for pods to access these node
+topology labels would provide users a straightforward, maintainable way to
+optimize their workloads given topology constraints.
+
+While "topology" is usually associated with the physical layout of a cluster,
+it can also be used to describe other types of information about the cluster.
+This KEP proposes to allow the expansion of the concept of topology to include
+user-defined aspects about their cluster nodes, and in turn provide a way for
+pods to receive this information.
+
+Workarounds today typically involve using an initContainer to query the
+Kubernetes API and then pass data via shared volume to other containers within
+the same pod. By comparison, this proposal would reduce the number of service
+accounts and API server clients. Another workaround is to use webhooks to inject
+labels into pods, but this relies on advance knowledge of where the pod is going
+to be scheduled and requires the webhook to be running and available at the time
+of pod creation. This proposal would provide an easier way to access node labels
+from pods, and would be more efficient than the current workarounds.
+
+### Goals
+
+* Values from labels `topology.k8s.io/*` and `kubernetes.io/hostname` are made
+  available via downward API 
+
+### Non-Goals
+
+* Not to expose additional node info outside of topology labels
+* Not to pass any additional node labels other than `topology.k8s.io/*` and
+  `*.topology.k8s.io/*` to pods
+* Not to update pod labels after the initial node -> pod copy has been made
+* Not to make assurances regarding timing and availability of the label beyond
+  the initial pod label copy at scheduling time
+* Not to make assurances about the immutability of the pod label after the
+  initial copy. As with other labels, the pod label can be updated by the user
+  after the pod is created.
+
+## Proposal
+
+KEP 1659 defines the following labels: `topology.kubernetes.io/region` and
+`topology.kubernetes.io/zone` to be used for topology information. These labels
+are useful for pods as well to be able to make application decisions based on
+the region or zone the pod is running in. This KEP proposes to make these labels
+available to pods while also expanding upon KEP 1659 to allow for user-defined
+labels in the `*.topology.kubernetes.io` namespace.
+
+This KEP expands on KEP 1659 in the following ways:
+
+1. Label prefixes of the form `<domain>.topology.kubernetes.io` are allocated
+   for use by end users. The Kubernetes project itself will not define any
+   labels with this prefix.
+2. Labels of the form `<domain>.topology.kubernetes.io/<field>` will be passed
+   to pods.
+3. Labels of the form `topology.kubernetes.io/*` will be passed to pods but will
```

Comment:
There was an open question about hostname?  In or out, we should say here.

Reply:
I think we should include it

Reply:
Main thing to consider is whether it's redundant with current field ref `spec.nodeName`. I believe there are some cases where `spec.nodeName` may differ from `kubernetes.io/hostname`, but examples are not coming to mind